utils/proc/ps_posix.go

```diff
@@ -36,3 +37,18 @@ func killGroup(ctx context.Context, pid int32) (err error) {
 	err = ConvertProcessError(syscall.Kill(-pgid, syscall.SIGKILL))
 	return
 }
+
+// WaitForCompletion will wait for a given process to complete.
+// This allows check to work if the underlying process was stopped without needing the os.Process that started it.
+func WaitForCompletion(ctx context.Context, pid int) (err error) {
+	pgid, err := syscall.Getpgid(pid)
```

Comment:
is there not a generic proc FindPGId or such a function in the process library in use uner the bonnet. If so, could we use it instead

Reply:
and could there be a helper for it?